ql/src/java/org/apache/hadoop/hive/ql/parse/UpdateDeleteSemanticAnalyzer.java

```diff
@@ -285,6 +297,58 @@ private void reparseAndSuperAnalyze(ASTNode tree, Table mTable, ASTNode tabNameN
     }
   }
 
+  private boolean tryMetadataUpdate(ASTNode tree, Table table, ASTNode tabNameNode) throws SemanticException {
+    TableName tableName = getQualifiedTableName(tabNameNode);
+    if (!deleting() || table.getStorageHandler() == null) {
+      return false;
+    }
+    int whereIndex = 1;
+    List<? extends Node> children = tree.getChildren();
+    if (children.size() <= whereIndex) {
+      return false;
+    }
+    ASTNode whereNode = (ASTNode) children.get(whereIndex);
+    String whereClause = ctx.getTokenRewriteStream().toString(
+        whereNode.getChild(0).getTokenStartIndex(), whereNode.getChild(0).getTokenStopIndex());
+    StringBuilder sb = new StringBuilder("insert into table ").append(getFullTableNameForSQL(tabNameNode))
```

Comment:
isn't it enough to have just select?

Reply:
btw, tasks ids in query plan are shifted and that looks a bit confusing

Reply:
Replaced it with a select query. 
Some Stage IDs have changed due to parsing twice.